### PR TITLE
STORE: Make it possible to tell the file loader to use secure memory

### PR DESCRIPTION
**Pull request by levitte**
This is an add-on to #3542, to be able to tell the 'file' scheme loader to use secure memory when possible.

Please review #3542 first.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -4,6 +4,22 @@
...
+     URI schemes.  There is one built in "loader" for the 'file' scheme.
+     [Richard Levitte]
+
+  *) Add a STORE module, which implements a uniform and URI based reader of
+     stores that can contain keys, certificates, CRLs and numerous other
+     objects.  The main API is loosely based on a few stdio functions, and
+     includes STORE_open, STORE_load, STORE_eof, STORE_error and STORE_close.
+     The implementation uses backends called "loaders" to implement arbitrary
+     URI schemes.  There is one built in "loader" for the 'file' scheme.
+     [Richard Levitte]
```
Comment (tmshort) [inline CHANGES]: comment added twice

**Comment by tmshort** [comment]
Copyrights need to be updated in various ways (likely just 2017?)

File: util/mkdef.pl [hunk 2]
```diff
@@ -284,13 +284,13 @@
...
 $crypto.=" include/openssl/ocsp.h";
 $crypto.=" include/openssl/ui.h";
-#$crypto.=" include/openssl/store.h";
 $crypto.=" include/openssl/cms.h";
 $crypto.=" include/openssl/srp.h";
 $crypto.=" include/openssl/modes.h";
 $crypto.=" include/openssl/async.h";
 $crypto.=" include/openssl/ct.h";
 $crypto.=" include/openssl/kdf.h";
+$crypto.=" include/openssl/store.h";
```
Comment (tmshort) [inline util/mkdef.pl]: This change is likely unnecessary.

Comment (levitte) [inline util/mkdef.pl]: Why?

**Comment by mattcaswell** [comment]
Can we rebase this?

**Comment by levitte** [comment]
Done.  I am in the process of doing so with all the remaining STORE PRs

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged. 7852f588a651ee30256678a23a6162a3b2fb822f

**Pull request closed without merging**